lti_consumer/lti.py

```diff
@@ -159,6 +159,32 @@ def get_signed_lti_parameters(self):
                 if len(language_preference) == 1:
                     self.xblock.user_language = language_preference[0].value
 
+            credit_service = self.xblock.runtime.service(self.xblock, 'credit')
+            user_id = getattr(real_user_object, "id", None)
+            if self.xblock.transmit_course_mode_and_status and credit_service and user_id:
```

Comment:
@robrap one slightly unfortunate thing here is that when the xBlock is loaded in Studio it does not have the full set of runtime support as it does when it is loaded into LMS. For example, we don't have access to a user_id, as I recall. I'll add a few logging items, but just because some of these settings might not be present in the runtime, it doesn't necessarily  make it an exception. But - as you say - more debugging information could be helpful for diagnostics.